### PR TITLE
[improvement](mtmv) Materialized view partition track supports date_trunc and optimize the fail reason

### PR DESCRIPTION
**Pull request by seawinde**
## Proposed changes

this depends on https://github.com/apache/doris/pull/34781

Materialized view partition track supports date_trunc and optimize the fail reason.
it supports create partition mv as following:
this mv will be partition updated by day

```
CREATE MATERIALIZED VIEW mv_6
BUILD IMMEDIATE REFRESH AUTO ON MANUAL
partition by(date_trunc(date_alias, 'day'))
DISTRIBUTED BY RANDOM BUCKETS 2
PROPERTIES ('replication_num' = '1')
AS
SELECT date_trunc(t1.L_SHIPDATE, 'hour') as date_alias, t2.O_ORDERDATE, t1.L_QUANTITY, t2.O_ORDERSTATUS, 
count(distinct case when t1.L_SUPPKEY > 0 then t2.O_ORDERSTATUS else null end) as cnt_1 
from 
  (select * from 
  lineitem 
  where L_SHIPDATE in ('2017-01-30')) t1 
left join 
  (select * from 
  orders 
  where O_ORDERDATE in ('2017-01-30')) t2 
on t1.L_ORDERKEY = t2.O_ORDERKEY 
group by 
t1.L_SHIPDATE, 
t2.O_ORDERDATE, 
t1.L_QUANTITY, 
t2.O_ORDERSTATUS;

```

### CODE REVIEW

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/Interval.java [hunk 1]
```diff
@@ -61,30 +63,51 @@ public <R, C> R accept(ExpressionVisitor<R, C> visitor, C context) {
...
      */
     public enum TimeUnit {
-        YEAR("YEAR", false),
-        MONTH("MONTH", false),
-        WEEK("WEEK", false),
-        DAY("DAY", false),
-        HOUR("HOUR", true),
-        MINUTE("MINUTE", true),
-        SECOND("SECOND", true);
+        YEAR("YEAR", false, 800),
```
[the remark this answers is not in the document]
Comment (seawinde) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/Interval.java:66]: Maybe will add new time unit in the future, maybe 100-800 keep the extensibility in the future

**Comment by seawinde** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by morrySnow** [comment]
run feut

**Comment by morrySnow** [comment]
run performance

**Review by morrySnow — DISMISSED** [review]

**Review by morrySnow — DISMISSED** [review]

**Comment by seawinde** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by morrySnow — DISMISSED** [review]

**Comment by seawinde** [comment]
run buildall

File: regression-test/suites/mtmv_p0/test_rollup_partition_mtmv.groovy [hunk 2]
```diff
@@ -420,4 +596,22 @@ suite("test_rollup_partition_mtmv") {
...
+        sql """
+        CREATE MATERIALIZED VIEW ${mvName}
+            BUILD IMMEDIATE REFRESH AUTO ON MANUAL
+            partition by (date_trunc(month_alias, 'hour'))
+            DISTRIBUTED BY RANDOM BUCKETS 2
+            PROPERTIES (
+            'replication_num' = '1'
+            )
+            AS
+            SELECT date_trunc(`k2`,'miniute') as month_alias, * FROM ${tableName};
```
Comment (zddr) [inline regression-test/suites/mtmv_p0/test_rollup_partition_mtmv.groovy:611]: miniute==>minute?month_alias==>miniute_alias?

File: regression-test/suites/mtmv_p0/test_rollup_partition_mtmv.groovy [hunk 3]
```diff
@@ -257,6 +374,65 @@ suite("test_rollup_partition_mtmv") {
...
+        sql """
+        CREATE MATERIALIZED VIEW ${mvName}
+            BUILD IMMEDIATE REFRESH AUTO ON MANUAL
+            partition by (date_trunc(`k2`, 'month'))
+            DISTRIBUTED BY RANDOM BUCKETS 2
+            PROPERTIES (
+            'replication_num' = '1'
+            )
+            AS
+            SELECT date_trunc(`k2`,'day') as month_alias FROM ${tableName};
```
Comment (zddr) [inline regression-test/suites/mtmv_p0/test_rollup_partition_mtmv.groovy:427]: month_alias==>day_alias?

**Comment by seawinde** [comment]
run buildall

File: regression-test/suites/mtmv_p0/test_rollup_partition_mtmv.groovy [hunk 4]
```diff
@@ -602,13 +602,13 @@ suite("test_rollup_partition_mtmv") {
...
             BUILD IMMEDIATE REFRESH AUTO ON MANUAL
-            partition by (date_trunc(month_alias, 'hour'))
+            partition by (date_trunc(miniute_alias, 'hour'))
             DISTRIBUTED BY RANDOM BUCKETS 2
             PROPERTIES (
             'replication_num' = '1'
             )
             AS
-            SELECT date_trunc(`k2`,'miniute') as month_alias, * FROM ${tableName};
+            SELECT date_trunc(`k2`,'miniute') as miniute_alias, * FROM ${tableName};
```
Comment (zddr) [inline regression-test/suites/mtmv_p0/test_rollup_partition_mtmv.groovy:611]: miniute==>minute?

**Review by zddr — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**